src/test/java/org/broadinstitute/hellbender/tools/spark/pipelines/MarkDuplicatesSparkIntegrationTest.java

```diff
@@ -187,6 +187,25 @@ public void testSupplementaryReadUnmappedMate() {
         }
     }
 
+    @Test
+    public void testHashCollisionHandling() {
+        File output = createTempFile("supplementaryReadUnmappedMate", "bam");
+        final ArgumentsBuilder args = new ArgumentsBuilder();
+        args.addOutput(output);
+        args.addInput(getTestFile("hashCollisionedReads.bam"));
+        runCommandLine(args);
+
+        try ( final ReadsDataSource outputReadsSource = new ReadsDataSource(output.toPath()) ) {
```

Comment:
Add a comment about this, it's not clear that this bam has reads that have a hash collision.